prow/monitoring/prometheus_expose.yaml

```diff
@@ -0,0 +1,32 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: prometheus
+  name: prometheus
+  namespace: prow-monitoring
+spec:
+  ports:
+  - name: http
+    port: 9090
+    protocol: TCP
+    targetPort: 9090
+  selector:
+    app: prometheus
+  sessionAffinity: ClientIP
+  type: ClusterIP
+---
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
```

Comment:
Ingress helps for debugging so that we can work with UI.

Is there any container which provides some authentication projection from the cluster? On openshift, we use [openshift/oauth-proxy](https://github.com/openshift/oauth-proxy)

Reply:
Commented out the `ingress` and leave it for the follow-up.
In the end, there will be 3 ingresses (or 3 URLs) for grafana/alertManager/prometheus.

In general, no authentication came with alertManager or prometheus out of box while grafana did.
If no authentication is eabled and we do not care about the whole world viewing the data:

* prometheus: [admin rest api is disabled](https://github.com/kubernetes/test-infra/blob/bb35e1f9ff41ca6a35de90b95d91e1e784a51225/prow/monitoring/prow_prometheus.yaml#L26) and our data are safe. Search UI is also open to the public (hard to say how harmful it is right now).
* alert manager: everyone could silence our alerts
* grafana: depends on how we set up our authentication here. In our production, every authenticated user is with read-only mode.

In my opinion, prometheus/alertmanager is open only to a restricted set of users and grafana can be viewed by anyone.